### PR TITLE
tests: fix and enable s3 snapshot test

### PR DESCRIPTION
**Pull request by andydunstall**
Fixes the S3 snapshot test and enables in regression tests.

### CODE REVIEW

**Comment by andydunstall** [comment]
Regression tests ran ok in https://github.com/dragonflydb/dragonfly/actions/runs/10898316504/job/30241308119

File: .github/actions/regression-tests/action.yml [hunk 1]
```diff
@@ -56,6 +62,12 @@ runs:
         if [[ $code -ne 0 ]]; then
           exit 1
         fi
+      env:
+        # Add environment variables to enable the S3 snapshot test.
+        DRAGONFLY_S3_BUCKET: dragonflydb-regression-tests
```
Comment (romange) [inline .github/actions/regression-tests/action.yml:67]: I would like to move it to secrets as well.

Comment (andydunstall) [reply .github/actions/regression-tests/action.yml]: sure updated

**Review by romange — DISMISSED** [review]

**Comment by andydunstall** [comment]
Oops looks like I configured the bucket wrong, will update

**Comment by andydunstall** [comment]
> Oops looks like I configured the bucket wrong, will update

Saying that, I can't find what I've done wrong. Just to check the secret is called `S3_REGTEST_BUCKET`?

**Comment by andydunstall** [comment]
Ah didn't notice CI calls regression test action not workflow, so the bucket is empty (as CI disables AWS updated to just not run the test if bucket empty)

**Comment by romange** [comment]
I did not understand the comment. Is it because we define it only from regression tests action and not in the CI?

**Comment by andydunstall** [comment]
Sorry, I mean CI sets `-DWITH_AWS:BOOL=OFF`, so assuming you want to keep that, then S3 tests won't work in CI (otherwise yeah we could set the environment variables in CI too)

They'll still run in regression tests

**Comment by romange** [comment]
Thanks Andy!

**Pull request merged**